modules/storages/lib/api/v3/storage_files/storage_folders_api.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module API
+  module V3
+    module StorageFiles
+      class StorageFoldersAPI < ::API::OpenProjectAPI
+        using ::Storages::Peripherals::ServiceResultRefinements
+
+        helpers ::Storages::Peripherals::StorageFileInfoConverter,
+                ::Storages::Peripherals::StorageErrorHelper
+
+        helpers do
+          def parent_path_from_parent_id(parent_id)
+            ::Storages::StorageFileService
+              .call(storage: @storage, user: current_user, file_id: parent_id)
+              .match(
+                on_success: lambda { |folder_info|
+                  path = URI.decode_uri_component(folder_info.location)
```

Comment:
Not sure if I am missing something here, but I needed to decode URI characters for the fetched information from the provider

Reply:
Also on OneDrive, `FileInfo#location` isn't what you would send up to `CreateFolder`, it should be `id`

Annoying? Yes. Terribly so.

Reply:
WDYT of moving this method to the create folder service under another name like `CreateFolderService.with_parent_id(folder_name, parent_id)`, so that the resource ends up with no special business logic?